### PR TITLE
chore: update golagci-lint to v1.158.0

### PR DESCRIPTION
**Pull request by lburgazzoli**
**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -73,3 +74,7 @@ linters:
     - whitespace
     - wrapcheck
     - wsl
+    - execinquery
+    # TODO: too much work at this stage, but they should be re-enabled
```
Comment (squakez) [inline .golangci.yml:78]: Just out of curiosity. What this comment exactly mean?

**Review by squakez — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (lburgazzoli) [reply .golangci.yml]: I have improved it a little bit the description

TL;DR; fixing the warning would have required to change a lot of files so I'll open some new issues to track the tasks once this gets merged.

**Comment by lburgazzoli** [comment]
@squakez do we have already a PR/commit for the security finding ?

**Comment by squakez** [comment]
> @squakez do we have already a PR/commit for the security finding ?

No, it should eventually fixed automatically when the golang action will move to `1.21.10`. Feel free to merge if that one is the only check failure you got.

**Pull request merged**